File: sponge-jython/examples/script/py/triggers.py
"""
Sponge Knowledge base
Triggers - Generating events and using triggers
"""

from java.util.concurrent.atomic import AtomicBoolean, AtomicInteger
from org.openksavi.sponge.examples import SampleJavaTrigger

def onInit():
    # Variables for assertions only
    sponge.setVariable("receivedEventA", AtomicBoolean(False))
    sponge.setVariable("receivedEventBCount", AtomicInteger(0))
    sponge.setVariable("receivedEventTestJavaCount", AtomicInteger(0))

class TriggerA(Trigger):
    def onConfigure(self):
        self.event = "a"
    def onRun(self, event):
        self.logger.debug("Received event: {}", event.name)
        sponge.getVariable("receivedEventA").set(True)

class TriggerB(Trigger):
    def onConfigure(self):
        self.event = "b"
    def onRun(self, event):
        self.logger.debug("Received event: {}", event.name)
        receivedEventBCount = sponge.getVariable("receivedEventBCount")
        if receivedEventBCount.get() == 0:
            self.logger.debug("Statistics: {}", sponge.statisticsSummary)
        receivedEventBCount.incrementAndGet()

def onLoad():
    sponge.enableJava(SampleJavaTrigger)

def onStartup():
    sponge.logger.debug("Startup {}, triggers: {}", sponge.info, sponge.engine.triggers)
    sponge.logger.debug("Knowledge base name: {}", sponge.kb.name)
    sponge.event("a").send()
    sponge.event("b").sendAfter(200, 200)
    sponge.event("testJava").send()

def onShutdown():
    sponge.logger.debug("Shutting down")

File: sponge-nashorn/examples/script/js/triggers.js
/**
 * Sponge Knowledge base
 * Triggers - Generating events and using triggers
 */

var AtomicBoolean = java.util.concurrent.atomic.AtomicBoolean;
var AtomicInteger = java.util.concurrent.atomic.AtomicInteger;

function onInit() {
    // Variables for assertions only
    sponge.setVariable("receivedEventA", new AtomicBoolean(false));
    sponge.setVariable("receivedEventBCount", new AtomicInteger(0));
    sponge.setVariable("receivedEventTestJavaCount", new AtomicInteger(0));
}

var TriggerA = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.withEvent("a");
    },
    onRun: function(self, event) {
        self.logger.debug("Received event: {}", event.name);
        sponge.getVariable("receivedEventA").set(true);
    }
});

var TriggerB = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.withEvent("b");
    },
    onRun: function(self, event) {
        self.logger.debug("Received event: {}", event.name);
        if (sponge.getVariable("receivedEventBCount").get() == 0) {
            self.logger.debug("Statistics: {}", sponge.statisticsSummary);
        }
        sponge.getVariable("receivedEventBCount").incrementAndGet();
    }
});

function onLoad() {
    sponge.enableJava(org.openksavi.sponge.examples.SampleJavaTrigger.class);
}

function onStartup() {
    sponge.logger.debug("Startup {}, triggers: {}", sponge.info, sponge.engine.triggers);
    sponge.logger.debug("Knowledge base name: {}", sponge.kb.name);
    sponge.event("a").send();
    sponge.event("b").sendAfter(200, 200);
    sponge.event("testJava").send();
}

function onShutdown() {
    sponge.logger.debug("Shutting down");
}
